sdk/resource/os_windows_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/windows/registry"
+
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+func TestPlatformOSDescription(t *testing.T) {
+	osDescription, err := resource.PlatformOSDescription()
+
+	require.NoError(t, err)
+	require.Regexp(t, `^(\w+\s)+\(\d+\)\s\[Version\s\d+(\.\d+){3}\]$`, osDescription)
+}
+
+func TestReadRegistryValues(t *testing.T) {
+	k, _ := registry.OpenKey(
+		registry.LOCAL_MACHINE, `SOFTWARE\Microsoft\Windows NT\CurrentVersion`, registry.QUERY_VALUE)
+
+	defer k.Close()
+
+	t.Run("ReadProductName", testReadProductName(t, k))
+	t.Run("ReadDisplayVersion", testReadDisplayVersion(t, k))
+	t.Run("ReadReleaseID", testReadReleaseID(t, k))
+	t.Run("ReadCurrentMajorVersionNumber", testReadCurrentMajorVersionNumber(t, k))
+	t.Run("ReadCurrentMinorVersionNumber", testReadCurrentMinorVersionNumber(t, k))
+	t.Run("ReadCurrentBuildNumber", testReadCurrentBuildNumber(t, k))
+	t.Run("ReadUBR", testReadUBR(t, k))
```

Comment:
nit: you could use a table-driven-test as you did in other places. e.g.:

```go
tt := []struct {
  Name          string
  RegKeyReader func(registry.Key) string
}

```

or maybe even inline everything like:
```go
assert.NotEmpty(resource.ReadProductName(key), "should read ProductName")
```

Reply:
I used to think of table-driven tests more as a way to test different inputs for the same function rather than testing different functions, but in a more general sense, it's true that they are more of a handy way to run subtests.

Liked your second approach. It's a good use for `assert`, which will execute all the assertions without failing the test, even if one doesn't match (ending up equivalent to subtests in this case).

Pushed here 983e5b35b7d6700890ffd1325e8de331e8105ae4.